### ISSUE TITLE
Allow our users to configure the compilation pipeline

### ISSUE THREAD

**Issue by sodic** [wasp-lang/wasp#866]
It would be nice if we allowed our users to configure their build pipeline. This includes:
- TSC configuration (done through `tsconfig.json`)
- NPM and Node configuration (done through `package.json`)
- Webpack configuration (maybe even choosing a bundler)
- etc.

Wasp would still come with sensible defaults, but could also allow overriding them. This is a pretty general issue and should be decomposed before we start working on it.

**Comment by Martinsos** [comment]
Yup! I am pretty sure we had old issue open for this but I just can't find it hm.
